pkg/render/dex.go

```diff
@@ -0,0 +1,350 @@
+	return &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      DexObjectName,
+			Namespace: DexNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"k8s-app": DexObjectName,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name: DexObjectName,
+					Port: DexPort,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: DexPort,
+					},
+					Protocol: corev1.ProtocolTCP,
+				},
+			},
+		},
+	}
+}
+
+// Perform a HTTP GET to determine if an endpoint is available.
+func (c *dexComponent) probe() *corev1.Probe {
+	return &corev1.Probe{
+		Handler: corev1.Handler{
+			HTTPGet: &corev1.HTTPGetAction{
+				Path:   "/dex/.well-known/openid-configuration",
+				Port:   intstr.FromInt(DexPort),
+				Scheme: corev1.URISchemeHTTPS,
+			},
+		},
+		InitialDelaySeconds: 90,
+		PeriodSeconds:       10,
+	}
+}
+
+func (c *dexComponent) configMap() *corev1.ConfigMap {
+	redirectURIs := []string{
+		"https://localhost:9443/login/oidc/callback",
+		"https://127.0.0.1:9443/login/oidc/callback",
+		"https://localhost:9443/tigera-kibana/api/security/oidc/callback",
+		"https://127.0.0.1:9443/tigera-kibana/api/security/oidc/callback",
+	}
+	host := c.dexConfig.ManagerURI()
+	if host != "" && !strings.Contains(host, "localhost") && !strings.Contains(host, "127.0.0.1") {
+		redirectURIs = append(redirectURIs, fmt.Sprintf("%s/login/oidc/callback", host))
+		redirectURIs = append(redirectURIs, fmt.Sprintf("%s/tigera-kibana/api/security/oidc/callback", host))
+	}
+
+	data := map[string]interface{}{
+		"issuer": fmt.Sprintf("%s/dex", c.dexConfig.ManagerURI()),
+		"storage": map[string]interface{}{
+			"type": "kubernetes",
+			"config": map[string]bool{
+				"inCluster": true,
+			},
+		},
+		"web": map[string]interface{}{
+			"https":                   "0.0.0.0:5556",
+			"tlsCert":                 "/etc/dex/tls/tls.crt",
+			"tlsKey":                  "/etc/dex/tls/tls.key",
+			"allowedOrigins":          []string{"*"},
+			"discoveryAllowedOrigins": []string{"*"},
+		},
+		"connectors": []map[string]interface{}{c.connector},
+		"oauth2": map[string]interface{}{
+			"skipApprovalScreen": true,
+			"responseTypes":      []string{"id_token", "code", "token"},
+		},
+		"staticClients": []map[string]interface{}{
+			{
+				"id":           DexClientId,
+				"redirectURIs": redirectURIs,
+				"name":         "Calico Enterprise Manager",
+				"secretEnv":    DexSecretEnv,
+			},
+		},
+	}
+
+	bytes, err := yaml.Marshal(data)
+	if err != nil { // Don't think this is possible.
+		panic(err)
+	}
+	return &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      DexObjectName,
+			Namespace: DexNamespace,
+		},
+		Data: map[string]string{
+			"config.yaml": string(bytes),
+		},
+	}
+}
+
+// This func prepares the configuration and objects that will be rendered related to the connector and its secrets.
+func getConnector(dexConfig DexConfig) map[string]interface{} {
```

Comment:
Everything in this could just be in the DexConfig. If this is the only thing you need the ConnectorType for then you could just have a GetConnector function on the DexConfig that creates and returns this map, it's all stuff specific to what's in the dex config.